src/Controller/QuestionTemplateController.php

```diff
@@ -64,23 +86,43 @@ public function createQuestion(Request $request): MessageInterface
      */
     public function showQuestions(Request $request, array $requestAttributes): Response
     {
+        $parameterBag = new ParameterBag([
+            'orderBy' => $request->getParameter('orderBy', ''),
+            'sort' => $request->getParameter('sort', ''),
+            'type' => $request->getParameter('type', ''),
+            'text' => $request->getParameter('text', ''),
+        ]);
+
+        $filters = [
+            'text' => $parameterBag->get('text'),
+            'type' => $parameterBag->get('type')
+        ];
+
         $resultsPerPage = 5;
-        $text = $this->questionTemplateService->getFromParameter('text', $request, "");
-        $type = $this->questionTemplateService->getFromParameter('type', $request, "");
+        //TODO remove casts
         $currentPage = (int)$this->questionTemplateService->getFromParameter('page', $request, 1);
-        $totalResults = (int)$this->questionTemplateService->getEntityNumberOfPagesByField(QuestionTemplate::class, ['type' => $type, 'text' => $text]);
-        $questions = $this->questionTemplateService->getEntitiesByField(QuestionTemplate::class, ['type' => $type, 'text' => $text], $currentPage, $resultsPerPage);
+        //TODO modify this method
+        $totalResults = (int)$this->questionTemplateService->getEntityNumberOfPagesByField(QuestionTemplate::class, $filters);
+        $filtersForEntity = new Filter(
+            $filters,
+            $resultsPerPage,
+            ($currentPage - 1) * $resultsPerPage,
+            $parameterBag->get('orderBy'),
+            $parameterBag->get('sort')
+        );
+        $questions = $this->repoManager->getRepository(QuestionTemplate::class)->getFilteredEntities($filtersForEntity);
 
         $paginator = new Paginator($totalResults, $currentPage, $resultsPerPage);
+        //TODO to be removed
         $paginator->setTotalPages($totalResults, $resultsPerPage);
 
         //TODO username modify after injecting the Session class in Controller
         return $this->renderer->renderView("admin-questions-listing.phtml", [
-            'text' => $text,
             'username' => $this->questionTemplateService->getName(),
-            'dropdownType' => $type,
             'paginator' => $paginator,
             'questions' => $questions,
+            'url' => $this->urlBuilder,
```

Comment:
inject UrlBuilder directly in the renderer, and pass it to the template from there